arch/xtensa/src/esp32/esp32_himem_chardev.c

```diff
@@ -0,0 +1,415 @@
+/****************************************************************************
+ * arch/xtensa/src/esp32/esp32_himem_chardev.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <debug.h>
+#include <sys/types.h>
+#include <nuttx/list.h>
+#include <nuttx/kmalloc.h>
+#include <nuttx/himem/himem.h>
+#include <nuttx/mutex.h>
+#include <nuttx/fs/fs.h>
+#include <arch/esp32/esp32_himem_chardev.h>
+#include "esp32_himem.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define HIMEM_UNMAPPED (-1)
+
+struct himem_chardev
+{
+  struct list_node node;
+  size_t size;
+  esp_himem_handle_t mem_handle;
+  char name[32];
+};
+
+struct himem_chardev_priv
+{
+  off_t offset; /* himem offset[byte]. update by seek(), read(), write(). */
+};
```

Comment:
Struct names not compliant to NuttX coding guidelines

Reply:
I agree.It's been revised.